src/_guides/openapi/advanced-ref-usage.md

```diff
@@ -0,0 +1,335 @@
+---
+title: "OpenAPI & AsyncAPI $ref: Advanced Guide"
+authors: Phil Sturgeon
+# image: images/guides/building-your-api-design-mode.png
+# canonical_url: https://bump.sh/blog/building-your-api-design-mode
+excerpt: Learn how to use $ref to reduce clutter and repetition in your OpenAPI and AsyncAPI documents.
+---
+
+After using OpenAPI or AsyncAPI for a while, you might notice your description documents have become a rather unwieldy mess of YAML and JSON. You end up with a whole lot of repetition, and the huge mess of YAML and JSON just loves trigger merge conflicts as multiple developers change different things but Git seems none the wiser.
+
+You can avoid this pain by splitting description documents up with `$ref`, putting various reusable components, but how exactly you go about doing that can be a tricky one to work out.
+
+## OpenAPI Reusable Components
+
+The [OpenAPI Documentation](https://learn.openapis.org/) includes a brilliant example of an API for playing the classic board game Tic Tac Toe. 
+
+This has several parts that are used several times, so instead of copy-pasting everything they've defined reusable `components` for `schemas` and `parameters`. 
+
+```yaml
+openapi: 3.1.0
+info:
+  title: Tic Tac Toe
+  description: |
+    This API allows writing down marks on a Tic Tac Toe board
+    and requesting the state of the board or of individual squares.
+  version: 1.0.0
+tags:
+  - name: Gameplay
+paths:
```

Comment:
I have a doubt whether we should include the whole YAML definition inside the blogpost, I understand we need to show the whole `paths` structure for later in the blog post, but what about keeping only the full definition of the first operation and hiding the content of all other operations? Something like:

```yaml
openapi: 3.1.0
info:
  title: Tic Tac Toe
  description: |
    This API allows writing down marks on a Tic Tac Toe board
    and requesting the state of the board or of individual squares.
  version: 1.0.0
tags:
  - name: Gameplay
paths:
  # Whole board operations
  /board:
    get:
      summary: Get the whole board
      description: Retrieves the current state of the board and the winner.
      tags:
        - Gameplay
      operationId: get-board
      responses:
        "200":
          description: "OK"
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/status"
  # Single square operations
  /board/{row}/{column}:
    parameters:
      - $ref: "#/components/parameters/rowParam"
      - $ref: "#/components/parameters/columnParam"
    get:
      # ... Hidden for readability...
    put:
      # ... Hidden for readability...
components:
  # ... Keep every components...